pkg/cmd/server/origin/master.go

```diff
@@ -273,6 +271,55 @@ func initAPIVersionRoute(root *restful.WebService, version string) {
 		Consumes(restful.MIME_JSON))
 }
 
+// If we know the location of the asset server, redirect to it when / is requested
+// and the Accept header supports text/html
+func assetServerRedirect(handler http.Handler, assetPublicURL string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		accept := req.Header.Get("Accept")
+		if req.URL.Path == "/" && strings.Contains(accept, "text/html") {
```

Comment:
@jwforres  I'd say this might be error prone, looking at [this http spec](http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html), which allows you to specify the accept header in quite sophisticated way. Eg. if I'd send `Accept: text/html; q=0.2, application/json` you'd classify this as web request, whereas actually `text/htm`l option has `q=0.2` and `application/json` has `q=1` which means is the preferred one. Other example would be that there is possibility the header will contain `*/*` which may go with `text/html` but will not with this. I couldn't find any go lib to cover this, but [here](https://github.com/mitsuhiko/werkzeug/blob/80d674429f3ea76a706e6dc2419a3cecf7b2a33b/werkzeug/datastructures.py#L1648-L1666) implementation of this in python. I'm willing to help with this or provide more details on it as I've spent quite some time digging into this some time ago. On the other hand maybe something has changed in the mean time.